lib/ash/resource/transformers/no_reserved_field_names.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Ash.Resource.Transformers.NoReservedFieldNames do
+  @reserved_names [
```

Comment:
This might be a bit of an unorthodox approach, but I wanted to have a source of truth that can be referenced in both the moduledocs and the error.

Reply:
https://i.ytimg.com/vi/LaiN63o_BxA/maxresdefault.jpg

Reply:
Its perfect, this is very much aligned with the way we do things with Ash :D